plugins/woocommerce/client/legacy/js/frontend/a8c-address-autocomplete-service.js

```diff
@@ -0,0 +1,459 @@
+				const cacheKey = `${ inputValue }:${ country }`;
+				return searchCache.get( cacheKey ) || null;
+			};
+
+			// Helper function to store result in cache
+			const cacheResult = ( inputValue, country, data ) => {
+				const cacheKey = `${ inputValue }:${ country }`;
+				searchCache.set( cacheKey, data );
+
+				// Clean up cache if it gets too large (more than 100 entries)
+				if ( searchCache.size > 100 ) {
+					// Remove oldest entries (first 20 entries)
+					const entries = Array.from( searchCache.entries() );
+					entries.slice( 0, 20 ).forEach( ( [ key ] ) => {
+						searchCache.delete( key );
+					} );
+				}
+			};
+
+			// Shared error handling function
+			const handleApiError = ( data, response ) => {
+				if ( ! data.code && ! data.error ) {
+					return null; // No error to handle
+				}
+
+				const errorCode = data.code || data.error;
+
+				switch ( errorCode ) {
+					case 'expired_jwt_token':
+					case 'malformed_jwt_token':
+					case 'invalid_jwt_token':
+					case 'invalid_issuer':
+					case 'invalid_service':
+					case 'missing_jwt_token':
+						permanentlyDisabledServices.push( key );
+						console.error(
+							'Automattic Address Suggestion has been disabled due to invalid JWT token'
+						);
+						return null;
+					case 'rate_limit_exceeded':
+						permanentlyDisabledServices.push( key );
+						setTimeout( () => {
+							permanentlyDisabledServices.splice(
+								permanentlyDisabledServices.indexOf( key ),
+								1
+							);
+						}, response.headers.get( 'RateLimit-Retry-After' ) * 1000 );
+						console.error(
+							'Automattic Address Suggestion has been disabled due to rate limit exceeded'
+						);
+						return null;
+					case 'missing_query':
+						return null;
+					case 'no_suggestions':
+						return [];
+					case 'missing_address_id':
+						console.error(
+							'Automattic Address Suggestion: Missing address ID'
+						);
+						return null;
+					case 'no_place':
+						console.error(
+							'Automattic Address Suggestion: No place found'
+						);
+						return null;
+					case 'missing_session_id':
+						sessionId =
+							crypto && crypto.randomUUID
+								? crypto.randomUUID()
+								: Math.random().toString( 36 ).substring( 2 );
+						return null;
+					case 'woo_address_suggestion_internal_error':
+					case 'woo_address_suggestion_service_error':
+					case 'woo_address_suggestion_server_error':
+						serviceErrorRetries++;
+						if (
+							serviceErrorRetries >= MAX_SERVICE_ERROR_RETRIES
+						) {
+							permanentlyDisabledServices.push( key );
+							console.error(
+								'Automattic Address Suggestion has been disabled due to internal service error'
+							);
+						}
+						return null;
+					default:
+						return null;
+				}
+			};
+
+			const debouncedSearch = debounce(
+				async ( inputValue, country ) => {
+					const params = new URLSearchParams( {
+						query: inputValue,
+						country,
+						lang: document.documentElement.lang,
```

Comment:
```suggestion
						lang: document.documentElement.lang || navigator.lang,
```
Maybe worth doing just in case?